### PR TITLE
incoming!

### PR DESCRIPTION
**Pull request by stbuehler**
:)

### CODE REVIEW

File: 99-antispam_with_sieve.conf [hunk 1]
```diff
@@ -17,46 +17,33 @@ plugin {
...
-  #
   # rspamd:
   #
-  # X-Spamd-Bar: /
-  # X-Spamd-Result: default: False [0.00 / 150.00];
+  # X-Spamd-Result: default: False [0.00 / 150.00]; ...
   # X-Spam: Yes
 
   ##
-  #
```
Comment (darix) [inline 99-antispam_with_sieve.conf:32]: Why did you delete those? can we restore the examples for SA?

File: learn-spam.sieve [hunk 2]
```diff
@@ -1,8 +1,4 @@
 require ["vnd.dovecot.pipe", "copy", "imapsieve", "environment", "variables", "imap4flags"];
 
-if environment :matches "imap.user" "*" {
```
Comment (darix) [inline learn-spam.sieve:3]: why do you remove the username part? does the $USER part also still work when you have e.g. all mails under an user "vmail" e.g.?

File: learn-spam.rspamd.script [hunk 3]
```diff
@@ -1,15 +1,86 @@
 #!/bin/bash
-RSPAMD_CONTROLLER_PASSWORD_FILE="/etc/dovecot/rspamd-controller-password"
```
Comment (darix) [inline learn-spam.rspamd.script:2]: I wonder if we should then rename the script to learn.rspamd.script and treat both the action based names as symlinks,

File: global-try-spam.sieve [hunk 4]
```diff
@@ -0,0 +1,9 @@
+require ["fileinto", "mailbox", "spamtest", "relational", "comparator-i;ascii-numeric"];
+
+# don't create INBOX/Spam; only sort mail if it already exists
+if mailboxexists "INBOX/Spam" {
```
Comment (darix) [inline global-try-spam.sieve:4]: tbh ... if you use the default config with the special-use mailboxes the folder should probably exist already.

and the reason why we enforced the sorting here, is to make the "learn as ham" part work ... otherwise one might have to introduce a "ham" mailbox just for training that. currently the whole train has ham workflow works with the admittedly opinionated defaults.

See also https://github.com/dovecot/core/blob/master/doc/example-config/conf.d/15-mailboxes.conf#L46..L65

File: 99-antispam_with_sieve.conf (hunk 1 above)
Comment (stbuehler) [reply 99-antispam_with_sieve.conf:32]: the project is named "dovecot...rspamd". unrelated examples are imho not useful. what do you need them for?

File: learn-spam.sieve (hunk 2 above)
Comment (stbuehler) [reply learn-spam.sieve:3]: the full commit 2cdf702403e084ef56bb4369d738eb166565bf24 should explain all this.

File: learn-spam.rspamd.script (hunk 3 above)
Comment (stbuehler) [reply learn-spam.rspamd.script:2]: but where to put that script? not in the sieve pipe directory, because it shouldn't get called directly.

anyway: with a separate file new questions arise: symlink, `source` or `exec` ? :)

anyway, that is imho a question that can be solved some other time.

File: global-try-spam.sieve (hunk 4 above)
Comment (stbuehler) [reply global-try-spam.sieve:4]: This is just an example file and not used in the provided config - I'll keep using it, and just wanted to provide it too, but I don't mind if it's not in the repo.

File: README.md [hunk 5]
```diff
@@ -27,18 +23,146 @@ https://wiki2.dovecot.org/Pigeonhole/Sieve/Extensions/SpamtestVirustest
...
+[`imap.user`] from the sieve environment.
+
+This is sadly not documented.
+
+[`imap.user`]: https://github.com/dovecot/pigeonhole/blob/43f5835b3830830cb84a04a5a06c7e6b15cc21df/src/plugins/imapsieve/ext-imapsieve-environment.c#L28
+[`$USER`]: https://github.com/dovecot/pigeonhole/blob/43f5835b3830830cb84a04a5a06c7e6b15cc21df/src/plugins/sieve-extprograms/sieve-extprograms-common.c#L547
+
+# Related bugs
+
+- "rspamc learn fails for already learned messages": https://github.com/rspamd/rspamd/issues/2691
```
Comment (c-rosenberg) [inline README.md:153]: https://github.com/rspamd/rspamd/issues/2691

Das sollte nur rspamd Installationen < 1.8.3 betreffen.

File: README.md [hunk 6]
```diff
@@ -27,18 +23,146 @@ https://wiki2.dovecot.org/Pigeonhole/Sieve/Extensions/SpamtestVirustest
...
+   mail a second time):
+
+    ```
+    dovecot[...]: imap(...): program `/.../learn-spam.rspamd.script' terminated with non-zero exit code 1
+    dovecot[...]: imap(...): Error: sieve: pipe action: failed to execute to program `learn-spam.rspamd.script': refer to server log for more information.   [YYYY-MM-DD HH:mm:ss]
+    dovecot[...]: imap(...): Error: sieve: Execution of script /.../learn-spam.sieve failed
+    ```
 
-2. set password in /etc/dovecot/rspamd-controller-password
+- *milter_headers: export add_header score*: https://github.com/rspamd/rspamd/issues/2699
```
Comment (c-rosenberg) [inline README.md:164]: https://github.com/rspamd/rspamd/issues/2699
